dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -0,0 +1,59 @@
+from datetime import datetime
+
+from dags.analyzer_helper.discord.transform_raw_data_base import TransformRawDataBase
+
+
+class DiscordTransformRawData(TransformRawDataBase):
+    def transform(self, raw_data: list, platform_id: str, period: datetime) -> list[dict]:
+        transformed_data = []
+        for data in raw_data:
+            interactions = []
+            if data.get("replied"):
+                interactions.append({
+                    "name": "reply",
+                    "users_engaged_id": [data.get("replied")],
+                    "type": "receiver"
+                })
+
+            if data.get("user_mentions"):
+                interactions.append({
+                    "name": "mention",
+                    "users_engaged_id": data.get("user_mentions"),
+                    "type": "emitter"
+                })
+
+            if data.get("reactions"):
+                for reaction in data.get("reactions"):
+                    parts = reaction.split(", ")
+                    if len(parts) > 2:
+                        interactions.append({
+                            "name": "reaction",
+                            "users_engaged_id": parts[:-1],
+                            "type": "emitter"
+                        })
+
+            action_name = "message"
+            action_type = "emitter" if not data.get("isGeneratedByWebhook", False) else "webhook"
+
+            transformed_item = {
+                "author_id": data.get("author", ""),
+                "date": period,
+                "source_id": platform_id,
+                "metadata": {
+                    "channel_id": data.get("channelId", ""),
+                    "channel_name": data.get("channelName", ""),
+                    "thread_id": data.get("threadId", ""),
+                    "thread_name": data.get("threadName", ""),
+                    "message_id": data.get("messageId", ""),
+                    "is_generated_by_webhook": data.get("isGeneratedByWebhook", False)
```

Comment:
All the fields `channelId`, `channelName`, `threadId`, `threadName`, `messageId`, and `isGeneratedByWebhook` shouldn't be get via the `get` method. Please get them by directly calling them like this

```
data["channelId"]
```

Reply:
We don't want to store the `message_id` in metadata. It would be in source id of the rawmemberactivities document. [ref link](https://rndadocs.notion.site/Analyzer-Discord-Pipeline-5009512d82a243deb35b3388e17b29c5#:~:text=%22source_id%22%3A%20string%2C)


I found out we didn't mention the metadata for discord in our documents. Updated it and now you can see what exactly we want to save in metadata of rawmemberactivities: here's the [link](https://rndadocs.notion.site/Analyzer-Discord-Pipeline-5009512d82a243deb35b3388e17b29c5#:~:text=The%20metadata%20we%20want%20to%20save%20for%20discord)